### ISSUE TITLE
Spacemacs: loosing copied content while moving to another file.

### ISSUE THREAD

**Issue by Antiarchitect** [Alexander-Miller/treemacs#335]
HOW TO REPRODUCE:
Copy a text block `y` in the opened buffer.
Move to another file via treemaps and open it in place of the previous buffer.
Messages buffer writes somewhat:
```
Saving file /home/andrey/.emacs.d/.cache/treemacs-persist...
Wrote /home/andrey/.emacs.d/.cache/treemacs-persist
```
Try to paste copied text block `p`:
You paste Project name.

Not sure if it can be reproduced 100% times, but appears really often and annoys hard

**Comment by Alexander-Miller** [comment]
Cannot reproduce. Some really weird is going on here.

> Saving file /home/andrey/.emacs.d/.cache/treemacs-persist...

That only happens if something changes with your projects, when one is added, removed or a position is changed.

> You paste Project name.

And that only happens when you call `treemacs-copy-project-root` which is bound to `yr`.

So yeah, what version of spacemacs and treemacs are you on, what keys do you press and what does your config look like.

**Comment by Antiarchitect** [comment]
https://github.com/Antiarchitect/.dotfiles/blob/master/.spacemacs
Spacemacs is latest `develop` branch treemacs layer updates with it.
It's enough to open file from one project, select and copy block with `y`, SPC-0 back to Treemacs, go to another project, open file with `o-o` in place of the old, and try to paste with `p`.

**Comment by Alexander-Miller** [comment]
Let's get a tie breaker, and ask other spacemacs+treemacs users: @duianto, @TimoFreiberg - can you guys do me the favor and see if you can reproduce this?

**Comment by duianto** [comment]
#### TLDR:
It seems to be working as expected.

#### Long version:
With two projects defined Treemacs (both cloned from their github repositories):
- `helm`
- `spacemacs-theme` 

#### Steps to reproduce
- Restart Spacemacs develop (rev. bd77a5df6) without resuming files/layouts `SPC q R`
- Open Treemacs `SPC f t`
- Open `helm-adaptive.el`
- Copy `yy` the first line:
```
;;; helm-adaptive.el --- Adaptive Sorting of Candidates. -*- lexical-binding: t -*-
```
- Select the Treemacs window `SPC 0`
- Select the next project (`spacemacs-theme`) `C-j`
- Center the selection in the Treemacs window `C-l` (Ctrl and lowercase L)
- Open `spacemacs-common.el` `o o`
- Paste `p`

#### Observed and Expected

The copied line from the first file is inserted after the second files first line
```
;;; spacemacs-common.el --- Color theme with a dark and light versions.
;;; helm-adaptive.el --- Adaptive Sorting of Candidates. -*- lexical-binding: t -*-
```

#### System Info :computer:
- OS: windows-nt
- Emacs: 26.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. bd77a5df6)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(autohotkey colors emacs-lisp git helm html ibuffer markdown multiple-cursors org spell-checking treemacs version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS THREADS LCMS2

**Comment by TimoFreiberg** [comment]
Can't reproduce either.
I used the same commands as you guys.
Emacs 26.1, Archlinux, Spacemacs develop

**Comment by Antiarchitect** [comment]
I'm using `l` (Lowercase L) to open subfolders and navigate into, `h` - to move to parent - seems habit from Neotree. Use `j` to move down, and `k` for up. Don't use `C-j` `C-l`

**Comment by Alexander-Miller** [comment]
> I'm using `l` (Lowercase L) to open subfolders and navigate into,

You should've said so from the start. With this info I've found the culprit. 

Lesson of the day: Don't use `kill-whole-line` if you don't want to pollute the kill-ring. Use something like `(delete-region (point-at-bol) (1+ (line-end-position)))` instead. 

I'll be dropping a patch tomorrow.

**Comment by duianto** [comment]
@Antiarchitect Welcome. Don't worry you'll learn the more you use it. 😺 

@Alexander-Miller The fix seems to be working, the `kill-ring` isn't modified after changing the root up or down. Thanks 👍
